src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,172 @@
+import { useSearchParams } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { getSearchWith } from '../utils/searchHelper';
+import { Person } from '../types';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  useEffect(() => {
+    setLoading(true);
+    getPeople()
+      .then(peopleFromServer => {
+        setPeople(peopleFromServer);
+        setHasError(false);
+      })
+      .catch(() => {
+        setHasError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const handleNameFilterChange = (query: string) => {
+    setSearchParams(getSearchWith(searchParams, { query: query || null }));
+  };
+
+  const handleCenturyFilterChange = (century: string, isSelected: boolean) => {
+    const currentCenturies = searchParams.getAll('centuries');
+    const updatedCenturies = isSelected
+      ? currentCenturies.filter(c => c !== century)
+      : [...currentCenturies, century];
+
+    setSearchParams(
+      getSearchWith(searchParams, {
+        centuries: updatedCenturies.length ? updatedCenturies : null,
+      }),
+    );
+  };
+
+  const handleSexFilterChange = (sex: string | null) => {
+    setSearchParams(getSearchWith(searchParams, { sex: sex || null }));
+  };
+
+  const handleSort = (field: string) => {
+    const currentSort = searchParams.get('sort');
+    const currentOrder = searchParams.get('order');
+
+    let newSort = field;
+    let newOrder = 'asc';
+
+    if (currentSort === field) {
+      if (currentOrder === 'asc') {
+        newOrder = 'desc';
+      } else if (currentOrder === 'desc') {
+        newSort = '';
+        newOrder = '';
+      }
+    }
+
+    setSearchParams(
+      getSearchWith(searchParams, {
+        sort: newSort || null,
+        order: newOrder || null,
+      }),
+    );
+  };
+
+  const filteredPeople = people.filter(person => {
+    const query = searchParams.get('query')?.toLowerCase() || '';
+    const centuries = searchParams.getAll('centuries').map(Number);
+    const sexFilter = searchParams.get('sex');
+    const matchesQuery = person.name.toLowerCase().includes(query);
+    const matchesCentury =
+      centuries.length === 0 ||
+      centuries.includes(Math.ceil(person.born / 100));
+    const matchesSex = !sexFilter || person.sex === sexFilter;
+
+    return matchesQuery && matchesCentury && matchesSex;
+  });
+
+  const sortedPeople = [...filteredPeople].sort((a, b) => {
```

Comment:
You can wrap it into useMemo to avoid recalculating it every time an unrelated state or search parameter changes. Same for filteredPeople